clients/src/main/java/org/apache/kafka/common/network/SSLTransportLayer.java

```diff
@@ -222,11 +222,13 @@ public void handshake() throws IOException {
                               channelId, appReadBuffer.position(), netReadBuffer.position(), netWriteBuffer.position());
                     handshakeResult = handshakeWrap(write);
                     if (handshakeResult.getStatus() == Status.BUFFER_OVERFLOW) {
-                        int currentPacketBufferSize = packetBufferSize();
-                        netWriteBuffer = Utils.ensureCapacity(netWriteBuffer, currentPacketBufferSize);
-                        if (netWriteBuffer.position() >= currentPacketBufferSize) {
+                        int currentNetWriteBufferSize = netWriteBufferSize();
+                        netWriteBuffer.position(netWriteBuffer.limit());
+                        netWriteBuffer = Utils.ensureCapacity(netWriteBuffer, currentNetWriteBufferSize);
+                        netWriteBuffer.flip();
+                        if (netWriteBuffer.position() >= currentNetWriteBufferSize) {
```

Comment:
Yes, you are right. Presumably that is an assertion to check that JRE SSLEngine is working as expected. It should really check `limit` rather than `position`. Will make that change.